src/Apps/W1/External Storage - Document Attachments/app/src/AutomaticSync/DAExternalStorageSync.Report.al

```diff
@@ -0,0 +1,174 @@
+                        Message(NoRecordsMsg);
+                    CurrReport.Break();
+                end;
+
+                if MaxRecordsToProcess > 0 then
+                    if TotalCount > MaxRecordsToProcess then
+                        TotalCount := MaxRecordsToProcess;
+
+                ProcessedCount := 0;
+                FailedCount := 0;
+                DeleteCount := 0;
+                DeleteFailedCount := 0;
+
+                if GuiAllowed() then
+                    Dialog.Open(ProcessingMsg, TotalCount);
+            end;
+
+            trigger OnAfterGetRecord()
+            begin
+                ProcessedCount += 1;
+
+                if GuiAllowed() then
+                    Dialog.Update(1, ProcessedCount);
+
+                case SyncDirection of
+                    SyncDirection::"To External Storage":
+
+                        if not ExternalStorageProcessor.UploadToExternalStorage(DocumentAttachment) then
+                            FailedCount += 1;
+                    SyncDirection::"From External Storage":
+
+                        if not ExternalStorageProcessor.DownloadFromExternalStorage(DocumentAttachment) then
+                            FailedCount += 1;
+                end;
+                if DeleteExpiredFiles then
+                    if CalcDate('<+' + GetDateFormulaFromExternalStorageSetup() + '>', DocumentAttachment."External Upload Date".Date()) >= Today() then
+                        if ExternalStorageProcessor.DeleteFromInternalStorage(DocumentAttachment) then
+                            DeleteCount += 1
+                        else
+                            DeleteFailedCount += 1;
+
+                if (MaxRecordsToProcess > 0) and (ProcessedCount >= MaxRecordsToProcess) then
+                    CurrReport.Break();
+            end;
+
+            trigger OnPostDataItem()
+            begin
+                if GuiAllowed() then begin
+                    if TotalCount <> 0 then
+                        Dialog.Close();
+                    if DeleteExpiredFiles then
+                        Message(DeletedExpiredFilesMsg, ProcessedCount - FailedCount, FailedCount, DeleteCount)
+                    else
+                        Message(ProcessedMsg, ProcessedCount - FailedCount, FailedCount);
+                end;
+            end;
+        }
+    }
+
+    requestpage
+    {
+        SaveValues = true;
+
+        layout
+        {
+            area(Content)
+            {
+                group(General)
+                {
+                    Caption = 'General';
+                    field(SyncDirectionField; SyncDirection)
+                    {
+                        ApplicationArea = Basic, Suite;
+                        Caption = 'Sync Direction';
+                        OptionCaption = 'To External Storage,From External Storage';
+                        ToolTip = 'Specifies whether to sync to external storage, from external storage, or delete expired files.';
```

Comment:
This ToolTip implies three values, but I can only see the in the OptionCaption?
Is there something missing?